perses/tests/test_relative.py

```diff
@@ -577,3 +577,273 @@ def test_HybridTopologyFactory_energies(molecule_perturbation_list = [['naphthal
+    geometry_engine = FFAllAngleGeometryEngine(metadata=None,
+                                               use_sterics=False,
+                                               n_bond_divisions=100,
+                                               n_angle_divisions=180,
+                                               n_torsion_divisions=360,
+                                               verbose=True,
+                                               storage=None,
+                                               bond_softening_constant=1.0,
+                                               angle_softening_constant=1.0,
+                                               neglect_angles=False,
+                                               use_14_nonbondeds=True)
+
+    # Create geometry proposal
+    new_positions, logp_proposal = geometry_engine.propose(topology_proposal, positions, beta,
+                                                                   validate_energy_bookkeeping=True)
+    logp_reverse = geometry_engine.logp_reverse(topology_proposal, new_positions, positions, beta,
+                                                validate_energy_bookkeeping=True)
+
+    if not topology_proposal.unique_new_atoms:
+        assert geometry_engine.forward_final_context_reduced_potential == None, f"There are no unique new atoms but the geometry_engine's final context reduced potential is not None (i.e. {self._geometry_engine.forward_final_context_reduced_potential})"
+        assert geometry_engine.forward_atoms_with_positions_reduced_potential == None, f"There are no unique new atoms but the geometry_engine's forward atoms-with-positions-reduced-potential in not None (i.e. { self._geometry_engine.forward_atoms_with_positions_reduced_potential})"
+        vacuum_added_valence_energy = 0.0
+    else:
+        added_valence_energy = geometry_engine.forward_final_context_reduced_potential - geometry_engine.forward_atoms_with_positions_reduced_potential
+
+    if not topology_proposal.unique_old_atoms:
+        assert geometry_engine.reverse_final_context_reduced_potential == None, f"There are no unique old atoms but the geometry_engine's final context reduced potential is not None (i.e. {self._geometry_engine.reverse_final_context_reduced_potential})"
+        assert geometry_engine.reverse_atoms_with_positions_reduced_potential == None, f"There are no unique old atoms but the geometry_engine's atoms-with-positions-reduced-potential in not None (i.e. { self._geometry_engine.reverse_atoms_with_positions_reduced_potential})"
+        subtracted_valence_energy = 0.0
+    else:
+        subtracted_valence_energy = geometry_engine.reverse_final_context_reduced_potential - geometry_engine.reverse_atoms_with_positions_reduced_potential
+
+    # Generate repartitioned htf at lambda = 0
+    zero_htf = RepartitionedHybridTopologyFactory(topology_proposal=topology_proposal,
+                          current_positions=positions,
```

Comment:
could we just have a helper function that can just be called because there is a bit of code duplication here with just a few kwargs changed